tests/common/snappi/snappi_fixtures.py

```diff
@@ -517,3 +521,361 @@ def cvg_api(snappi_api_serv_ip,
     yield api
     if getattr(api, 'assistant', None) is not None:
         api.assistant.Session.remove()
+
+
+def snappi_dut_base_config(duthost_list,
+                           tgen_ports,
+                           snappi_ports,
+                           snappi_api):
+    """
+    Geenrate snappi API config and port config information for the testbed
+    Args:
+        conn_graph_facts (pytest fixture)
+        fanout_graph_facts (pytest fixture)
+        duthosts (pytest fixture): list of DUTs
+        rand_one_dut_hostname (pytest fixture): DUT hostname
+        snappi_api(pytest fixture): Snappi API fixture
+    Returns:
+        - config (obj): Snappi API config of the testbed
+        - port_config_list (list): list of port configuration information
+    """
+
+    """ Generate L1 config """
+
+    config = snappi_api.config()
+    new_snappi_ports = [dict(list(sp.items()) + [('port_id', i)])
+                        for i, sp in enumerate(snappi_ports) if sp['location'] in tgen_ports]
+    pytest_assert(len(set([sp['speed'] for sp in new_snappi_ports])) == 1, 'Ports have different link speeds')
+    [config.ports.port(name='Port {}'.format(sp['port_id']), location=sp['location']) for sp in new_snappi_ports]
+    speed_gbps = int(new_snappi_ports[0]['speed'])/1000
+
+    config.options.port_options.location_preemption = True
+    l1_config = config.layer1.layer1()[-1]
+    l1_config.name = 'L1 config'
+    l1_config.port_names = [port.name for port in config.ports]
+    l1_config.speed = 'speed_{}_gbps'.format(speed_gbps)
+    l1_config.ieee_media_defaults = False
+    l1_config.auto_negotiate = False
+    l1_config.auto_negotiation.link_training = False
+    l1_config.auto_negotiation.rs_fec = True
+
+    pfc = l1_config.flow_control.ieee_802_1qbb
+    pfc.pfc_delay = 0
+    [setattr(pfc, 'pfc_class_{}'.format(i), i) for i in range(8)]
+
+    port_config_list = []
+
+    for index, duthost in enumerate(duthost_list):
+        config_result = __intf_config_multidut(
+                                                config=config,
+                                                port_config_list=port_config_list,
+                                                duthost=duthost,
+                                                snappi_ports=new_snappi_ports)
+        pytest_assert(config_result is True, 'Fail to configure Vlan interfaces')
+
+    return config, port_config_list, new_snappi_ports
+
+
+@pytest.fixture(scope="function")
+def get_multidut_snappi_ports(duthosts, conn_graph_facts, fanout_graph_facts):            # noqa: F811
+    """
+    Populate tgen ports and connected DUT ports info of T0 testbed and returns as a list
```

Comment:
why tgen?

Reply:
this is just a naming that has been used, no specific reason to it